### DISCUSSION TITLE
MO Field Slips

### DISCUSSION THREAD

**Discussion by mo-nathan** [MushroomObserver/mushroom-observer#1896] [Ideas]
An important process that happens at mycology related events is vouchering.  This typically involves giving each participant a small pad of paper forms that they are expected to either fill out in the field or once they get back to the identification area with their collections.  The most important information for the field slip is what day the collection was made, where it was made (often a foray/walk number or a locality near where the event happens), and who made the collection.  These days field slips typically have a foray code and a unique number.  For example, NAMA-NC23-1729 or TMF-2022-248.  There are also often boxes for adding online identifiers such as the observation number for either MO or iNat.  Additional information typically on these forms include a name, the initials of the identifier (typically an event mycologist), habitat checkboxes, and a place for notes.

Examples: https://mushroomobserver.org/images/1592453 (NAMA-NC23-1729) and https://mushroomobserver.org/images/1486355 (TMF-2022-248)

At the 2022 Telluride Mushroom Festival, Andy Wilson mentioned that providing a more reliable way to get from the field slip to the online observation would be a significant improvement over the current way that field slips are processed.  I played a bit with trying to read field slips like the TMF one as they are and was not particularly successful.  I think it could be achieved with optical character recognition and/or barcode reading software, but it wasn't straight forward.  We talked a bit about maybe using QR codes, but I didn't really pursue it at the time.

I recently realized that QR codes could in fact very neatly solve the problem.  If the QR code was of an MO URL that included the for code and the unique number, we could use it to first create an object in MO (let's call it a FieldSlip) and an associated MO Observation.  If such objects already exist, then the QR code could take you to the appropriate page.  This would eliminate the need for writing down the MO id entirely.  The QR code on the field slip could be used later to get back to the online observation.  We could also add the ability to search for the foray identifier from the MO search box.

The huge advantage of QR code is that most mobile phones can automatically read them either from the live camera or from a stored image.  In addition they can be easily generated by MO.  It would also be relatively easy to add a field slip generator to MO.  I've experimented with creating a QR code that contains the URL https://mushroomobserver.org/field_slip/nemf-2024-123.  It is readable from printed page using less than a 1 inch x 1 inch square.  Reading QR codes from stored images in the MO backend is not simple, but there are good JavaScript libraries that could be utilized to read them as they get uploaded into MO.

Once that functionality exists, we could expand the functionality to allow users to directly fill out additional field slip information directly in MO.  This could avoid having to have event mycologist write down ids with pen or pencil helping to make the data more readable and no longer requiring the event recording from transcribing the information.  If the code is included with any DNA sample that is taken, this would provide an easy reliable way to find the appropriate place to record the results and get to any potentially relevant information about the collection.

Another interesting question is how the system should act if a field slip from an event is used after the event.  I think this could fold into the existing MO Project work where project admins could turn off the ability for non-admins to add field references and observations to a project.  We should also give the user the option to not add the resulting field slip reference and observation to any associated project.

**Comment by mo-nathan** [comment]
Note that to add additional voucher information we would need to add a tag, value type of functionality to projects.  For example, "Foray #": ["Joe's Creek", "Lake Nimmo", "Mt. Jason"]
Then the voucher would have a "Foray #" field that folks could add an appropriate selection from.  I wonder if there's a reasonable rails gem for giving users a simple form building UI.

**Comment by mo-nathan** [comment]
Decided to change the overal concept from "Voucher Slip" to "Field Slip".

**Comment by mo-nathan** [comment]
Realized yesterday that FieldSlips as currently conceived overlap heavily with the existing CollectionNumbers features.  The most significant difference is that FieldSlips can have at most one Observation whereas there is a many-to-many relationship between CollectionNumbers and Observations.  Current stats on CollectionNumbers:

Total: 12773
Distinct collection_number_ids: 12752
Distinct observation_ids: 12491

A good example of an observation with multiple CollectionNumbers: https://mushroomobserver.org/546746.  It has both a number for SOMA Camp and for iNat.  The most extreme example at the moment is: https://mushroomobserver.org/317677.

The most extreme example of a CollectionNumber with multiple Observations: https://mushroomobserver.org/collection_numbers/5636.  These are a set of puffball observations for various different species.  This seems like it should be Project.

What needs to get resolved to merge the two concept is what to do with the CollectionNumbers with multiple Observations.  Creating distinct Projects for each case would be the simplest thing to do.  However, I'm not sure this make sense for cases where there are just two Observations.  Here's a distribution of the number of CollectionNumbers by the number of associated observation:
```
mysql> select count(*) collection_numbers, c observations from (select count(*) c, collection_number_id from observation_colle
+--------------------+--------------+
| collection_numbers | observations |
+--------------------+--------------+
|              12590 |            1 |
|                122 |            2 |
|                 16 |            3 |
|                  8 |            4 |
|                  3 |            5 |
|                  2 |            7 |
|                  2 |            8 |
|                  3 |            9 |
|                  1 |           10 |
|                  1 |           13 |
|                  1 |           15 |
|                  2 |           16 |
|                  1 |           25 |
+--------------------+--------------+
13 rows in set (0.02 sec)
```

**Comment by mo-nathan** [comment]
Review of CollectionNumbers with the most Observations:
Steph Jarvis SSJ - personal collection at the HDTH-SFSU, 25 obsies, https://mushroomobserver.org/collection_numbers/5636, Nice collection of observations related to puffballs. Should be a project
Ann Cavanaugh AC3, 16 obsies , https://mushroomobserver.org/collection_numbers/9852, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Ann Cavanaugh AC1, 16 obsies, https://mushroomobserver.org/collection_numbers/9851, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Damian chalvignac 1, 15 obsies, https://mushroomobserver.org/collection_numbers/829, No clear purpose.  Seems like a "species list" at best.
Paul Sadowski PS1, 13 obsies, https://mushroomobserver.org/collection_numbers/9848, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Ann Cavanaugh AC4, 10 obsies, https://mushroomobserver.org/collection_numbers/9914, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Ann Cavanaugh AC6, 9 obsies, https://mushroomobserver.org/collection_numbers/9921, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Ann Cavanaugh AC2, 9 obsies, https://mushroomobserver.org/collection_numbers/9849, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Paul Sadowski PS2, 9 obsies, https://mushroomobserver.org/collection_numbers/9681, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Ann Cavanaugh AC5, 8 obsies, https://mushroomobserver.org/collection_numbers/9931, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Paul Sadowski PS3, 8 obsies, https://mushroomobserver.org/collection_numbers/9882, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Paul Sadowski PS5, 7 obsies, https://mushroomobserver.org/collection_numbers/9933, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
LionsMane 11-11, 7 obsies, https://mushroomobserver.org/collection_numbers/2791, No clear purpose.  Seems like a "species list" at best.  LionsMane is the user.
Oluna & Adolf Ceska s.n., 5 obsies, https://mushroomobserver.org/collection_numbers/1886, No clear purpose.  Seems like a "species list" at best.
Knoughtylog Damon Filbert #1, 4 obsies, https://mushroomobserver.org/collection_numbers/10932, No clear purpose.  All appear to be duplicates of each other.
Ann Cavanaugh AC8, 4 obsies, https://mushroomobserver.org/collection_numbers/9855, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Paul Sadowski PS9, 4 obsies, https://mushroomobserver.org/collection_numbers/9886, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Paul Sadowski PS10, 4 obsies, https://mushroomobserver.org/collection_numbers/9932, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Paul Sadowski PS14, 4 obsies, https://mushroomobserver.org/collection_numbers/9944, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
Paul Sadowski PS8, 4 obsies, https://mushroomobserver.org/collection_numbers/9946, No clear purpose.  Seems like a "species list" at best.  Associated with Paul Sadowski
John W. Allen P. samuiensis, 4 obsies, https://mushroomobserver.org/collection_numbers/7498, No clear purpose.  All appear to be duplicates of each other.
O.EL O.EL 19, 4 obsies, https://mushroomobserver.org/collection_numbers/2980, No clear purpose.  All appear to be duplicates of each other.

As yet unreviewed:
```
mysql> Select tmp.count, cn.id, cn.name, cn.number from ( select count(*) count, collection_number_id from observation_collection_numbers group by collection_number_id) tmp, collection_numbers cn where tmp.collection_number_id = cn.id and tmp.count > 1 order by tmp.count;
+-------+-------+-------------------------------------+--------------------------------------------+
| count | id    | name                                | number                                     |
+-------+-------+-------------------------------------+--------------------------------------------+
|     2 |     5 | Bill Sheehan                        | 3503                                       |
|     2 |    10 | Jason Bolin                         | JAB88                                      |
|     2 |   131 | Joan Knapp                          | MO-298729                                  |
|     2 |   469 | Crystal                             | #2018-01                                   |
|     2 |   494 | Crystal                             | #2015-28                                   |
|     2 |   554 | Crystal                             | #2018-02                                   |
|     2 |   555 | Crystal                             | #2018-03                                   |
|     2 |   564 | Jonathan Frank                      | JLF5939                                    |
|     2 |   567 | Robert van der Horst                | 15-03-18-PCI                               |
|     2 |   675 | David Arora                         | Arora 18008                                |
|     2 |   795 | JandewCollins                       | 5                                          |
|     2 |   957 | Doc Lingo                           | MO#317458                                  |
|     2 |  1012 | Regina Johnson                      | 05733                                      |
|     2 |  1454 | Stuart Ingalls                      | 3137311377                                 |
|     2 |  1719 | Bill Sheehan                        | 5109                                       |
|     2 |  2045 | Jim                                 | 1                                          |
|     2 |  2185 | Jay Justice                         | MF #68012                                  |
|     2 |  2310 | Kingman Bond Graham                 | 76551                                      |
|     2 |  2328 | Jason Dart                          | 1100                                       |
|     2 |  2429 | BGR1017                             | BMF-17B                                    |
|     2 |  2475 | Norton                              | D-6                                        |
|     2 |  2488 | O.EL                                | O.EL 63                                    |
|     2 |  2499 | VSUfungi                            | BBD 151                                    |
|     2 |  2756 | VSUfungi                            | BBD 340                                    |
|     2 |  2941 | Alden Dirks                         | MF79061                                    |
|     2 |  2969 | David                               | 1                                          |
|     2 |  3121 | Palestine Museum of Natural History | 1014                                       |
|     2 |  3129 | Palestine Museum of Natural History | 1013                                       |
|     2 |  3135 | Palestine Museum of Natural History | 1011                                       |
|     2 |  3138 | Palestine Museum of Natural History | 1006                                       |
|     2 |  3161 | Palestine Museum of Natural History | 1065                                       |
|     2 |  3170 | Palestine Museum of Natural History | 1069                                       |
|     2 |  3535 | Reid Moran                          | 15454                                      |
|     2 |  3619 | Kathleen Dobson                     | 19-004                                     |
|     2 |  3666 | Terri Clements/Donna Fulton         | TAC 1693                                   |
|     2 |  3988 | Bill Sheehan                        | 5152                                       |
|     2 |  5335 | Noah Topping                        | 95191                                      |
|     2 |  5361 | Junojack                            | 0906                                       |
|     2 |  5391 | Cody White                          | 3                                          |
|     2 |  5442 | Ian                                 | Sept20 afternoon                           |
|     2 |  5608 | Andrew Khitsun                      | 123                                        |
|     2 |  5875 | Rachel                              | 00019                                      |
|     2 |  6024 | M.rial                              | 131                                        |
|     2 |  6143 | Huafang                             | 392                                        |
|     2 |  6161 | Rhonda Kotelko                      | RK2340                                     |
|     2 |  6279 | Douglas E Williams                  | 008                                        |
|     2 |  6286 | Ryan                                | 1                                          |
|     2 |  6317 | Luke Smithson                       | Obs number                                 |
|     2 |  6416 | SOMA Camp                           | SOMA2020-001                               |
|     2 |  6924 | pynklynx                            | 3                                          |
|     2 |  7068 | Michael Stanton                     | 2020050301                                 |
|     2 |  7119 | Robert Benson                       | RHB05172020-A                              |
|     2 |  7139 | CARTER                              | 52420-5                                    |
|     2 |  7216 | Ty Pi                               | TRP 207                                    |
|     2 |  7233 | Rootsrocker                         | Roots                                      |
|     2 |  7273 | CHUNKY                              | 1                                          |
|     2 |  7299 | Eddee                               | B1/                                        |
|     2 |  7450 | Ty Pi                               | TRP 256                                    |
|     2 |  7527 | Michael Stanton                     | 2020090504                                 |
|     2 |  7574 | Spike                               | MF 61764                                   |
|     2 |  7623 | Jmangs                              | P1                                         |
|     2 |  7626 | Robert Benson                       | B09172020b                                 |
|     2 |  7770 | Azatlani Azatlan                    | 0001-10/5/2020                             |
|     2 |  7909 | Jason Dart                          | 1713                                       |
|     2 |  7991 | Ozeda                               | 00-001                                     |
|     2 |  8042 | Jason C. Slot                       | JCS091420_A                                |
|     2 |  8204 | Aidan Marshall                      | 01                                         |
|     2 |  8239 | Michael Stanton                     | 2021090301                                 |
|     2 |  8252 | Darvin DeShazer                     | DD725                                      |
|     2 |  8280 | Milo                                | 1                                          |
|     2 |  8551 | Juan Carlos Mejía                   | 21-001                                     |
|     2 |  8623 | Vihane                              | 250720210012                               |
|     2 |  8629 | Ronald Brey                         | 1                                          |
|     2 |  8761 | Peter & Colleen Vachuska            | V057                                       |
|     2 |  9142 | Zachary Dennis                      | 21-001                                     |
|     2 |  9356 | Lori                                | 9                                          |
|     2 |  9426 | Tim moe                             | 2069546752                                 |
|     2 |  9436 | Michael Stanton                     | 2021112902                                 |
|     2 |  9528 | Michael Stanton                     | 2021121402                                 |
|     2 |  9679 | Joe Alec King                       | 01-30-22-4                                 |
|     2 |  9844 | Charles Luce                        | CL3                                        |
|     2 |  9845 | Charles Luce                        | CL1                                        |
|     2 |  9861 | Sara David Rosenbaum                | SDR13                                      |
|     2 |  9883 | Sara David Rosenbaum                | SDR4                                       |
|     2 |  9888 | Sara David Rosenbaum                | SDR5                                       |
|     2 |  9920 | Christine Jordan                    | CJ13                                       |
|     2 |  9923 | Nancy Slowik                        | NS3                                        |
|     2 |  9924 | Ann Cavanaugh                       | AC12                                       |
|     2 |  9925 | Nancy Slowik                        | NS4                                        |
|     2 |  9930 | Ann Cavanaugh                       | AC9                                        |
|     2 |  9943 | Sara David Rosenbaum                | SDR1                                       |
|     2 |  9949 | Louise Shirley                      | LS2                                        |
|     2 |  9953 | Christine Jordan                    | CJ3                                        |
|     2 |  9954 | Christine Jordan                    | CJ6                                        |
|     2 |  9964 | Gary Lincoff                        | GL2                                        |
|     2 |  9998 | Ann Cavanaugh                       | AC10                                       |
|     2 |  9999 | Paul Sadowski                       | PS7                                        |
|     2 | 10020 | Sara David Rosenbaum                | SDR6                                       |
|     2 | 10039 | Joe Alec King                       | 03-20-22-1                                 |
|     2 | 10082 | Jenifer Nina Burghardt              | John Burghardt                             |
|     2 | 10181 | Michael W. Beug                     | 02mwb042222                                |
|     2 | 10225 | Michael Beug                        | 01mwb050422                                |
|     2 | 10267 | WookieXXXL                          | Steven                                     |
|     2 | 10317 | Josh McGinnis                       | EMB-73                                     |
|     2 | 10399 | ItsAmy                              | AK0007                                     |
|     2 | 10416 | ItsAmy                              | AK0021                                     |
|     2 | 10556 | rho-marascena                       | 1                                          |
|     2 | 10719 | Sarah Culliton                      | 0024                                       |
|     2 | 10728 | Joseph D. Cohen                     | FunDIS 54263                               |
|     2 | 10747 | Django Grootmyers                   | DG22070910                                 |
|     2 | 10850 | Sarah Culliton                      | 0042                                       |
|     2 | 11049 | Cape Town Exotic Mushrooms          | Byron Laurens                              |
|     2 | 11067 | ItsAmy                              | AK0039                                     |
|     2 | 11182 | Mike Dechter                        | MD25                                       |
|     2 | 11245 | Spencer Wimmer                      | SW21                                       |
|     2 | 11288 | Fungi of the Claremont Colleges     | 5                                          |
|     2 | 11534 | Joe Mat                             | 3473                                       |
|     2 | 11671 | jimmystclaire                       | RF-002                                     |
|     2 | 11756 | jamood@hotmail.co.uk                | Agrocybe                                   |
|     2 | 11931 | nanoPore                            | S23-190                                    |
|     2 | 11957 | Happihyphae                         | I Naturalist ID: 178278397                 |
|     2 | 13062 | Steve                               | Not yet determined                         |
|     3 |     8 | Jason Bolin                         | JAB100                                     |
|     3 |  2282 | Gawoodlurker                        | 01-G.FV                                    |
|     3 |  3123 | Palestine Museum of Natural History | 1020                                       |
|     3 |  3124 | Palestine Museum of Natural History | 1007                                       |
|     3 |  3133 | Palestine Museum of Natural History | 1003                                       |
|     3 |  6354 | Antoniahansa                        | 5                                          |
|     3 |  7601 | Theresa Kadish                      | 1                                          |
|     3 |  9859 | Sara David Rosenbaum                | SDR9                                       |
|     3 |  9863 | Sara David Rosenbaum                | SDR2                                       |
|     3 |  9967 | Paul Sadowski                       | PS6                                        |
|     3 | 10252 | Cosmic Camote                       | 1                                          |
|     3 | 10269 | Joe Alec King                       | 05-29-22-1                                 |
|     3 | 10273 | Joe Alec King                       | 06-03-22-1                                 |
|     3 | 10341 | Colin Smith                         | hitch937@gmail.com                         |
|     3 | 10555 | rho                                 | 1                                          |
|     3 | 12851 | Chasity Self                        | 1                                          |
```

**Comment by mo-nathan** [comment]
URLs for the above:
https://mushroomobserver.org/collection_numbers/5
https://mushroomobserver.org/collection_numbers/10
https://mushroomobserver.org/collection_numbers/131
https://mushroomobserver.org/collection_numbers/469
https://mushroomobserver.org/collection_numbers/494
https://mushroomobserver.org/collection_numbers/554
https://mushroomobserver.org/collection_numbers/555
https://mushroomobserver.org/collection_numbers/564
https://mushroomobserver.org/collection_numbers/567
https://mushroomobserver.org/collection_numbers/675
https://mushroomobserver.org/collection_numbers/795
https://mushroomobserver.org/collection_numbers/957
https://mushroomobserver.org/collection_numbers/1012
https://mushroomobserver.org/collection_numbers/1454
https://mushroomobserver.org/collection_numbers/1719
https://mushroomobserver.org/collection_numbers/2045
https://mushroomobserver.org/collection_numbers/2185
https://mushroomobserver.org/collection_numbers/2310
https://mushroomobserver.org/collection_numbers/2328
https://mushroomobserver.org/collection_numbers/2429
https://mushroomobserver.org/collection_numbers/2475
https://mushroomobserver.org/collection_numbers/2488
https://mushroomobserver.org/collection_numbers/2499
https://mushroomobserver.org/collection_numbers/2756
https://mushroomobserver.org/collection_numbers/2941
https://mushroomobserver.org/collection_numbers/2969
https://mushroomobserver.org/collection_numbers/3121
https://mushroomobserver.org/collection_numbers/3129
https://mushroomobserver.org/collection_numbers/3135
https://mushroomobserver.org/collection_numbers/3138
https://mushroomobserver.org/collection_numbers/3161
https://mushroomobserver.org/collection_numbers/3170
https://mushroomobserver.org/collection_numbers/3535
https://mushroomobserver.org/collection_numbers/3619
https://mushroomobserver.org/collection_numbers/3666
https://mushroomobserver.org/collection_numbers/3988
https://mushroomobserver.org/collection_numbers/5335
https://mushroomobserver.org/collection_numbers/5361
https://mushroomobserver.org/collection_numbers/5391
https://mushroomobserver.org/collection_numbers/5442
https://mushroomobserver.org/collection_numbers/5608
https://mushroomobserver.org/collection_numbers/5875
https://mushroomobserver.org/collection_numbers/6024
https://mushroomobserver.org/collection_numbers/6143
https://mushroomobserver.org/collection_numbers/6161
https://mushroomobserver.org/collection_numbers/6279
https://mushroomobserver.org/collection_numbers/6286
https://mushroomobserver.org/collection_numbers/6317
https://mushroomobserver.org/collection_numbers/6416
https://mushroomobserver.org/collection_numbers/6924
https://mushroomobserver.org/collection_numbers/7068
https://mushroomobserver.org/collection_numbers/7119
https://mushroomobserver.org/collection_numbers/7139
https://mushroomobserver.org/collection_numbers/7216
https://mushroomobserver.org/collection_numbers/7233
https://mushroomobserver.org/collection_numbers/7273
https://mushroomobserver.org/collection_numbers/7299
https://mushroomobserver.org/collection_numbers/7450
https://mushroomobserver.org/collection_numbers/7527
https://mushroomobserver.org/collection_numbers/7574
https://mushroomobserver.org/collection_numbers/7623
https://mushroomobserver.org/collection_numbers/7626
https://mushroomobserver.org/collection_numbers/7770
https://mushroomobserver.org/collection_numbers/7909
https://mushroomobserver.org/collection_numbers/7991
https://mushroomobserver.org/collection_numbers/8042
https://mushroomobserver.org/collection_numbers/8204
https://mushroomobserver.org/collection_numbers/8239
https://mushroomobserver.org/collection_numbers/8252
https://mushroomobserver.org/collection_numbers/8280
https://mushroomobserver.org/collection_numbers/8551
https://mushroomobserver.org/collection_numbers/8623
https://mushroomobserver.org/collection_numbers/8629
https://mushroomobserver.org/collection_numbers/8761
https://mushroomobserver.org/collection_numbers/9142
https://mushroomobserver.org/collection_numbers/9356
https://mushroomobserver.org/collection_numbers/9426
https://mushroomobserver.org/collection_numbers/9436
https://mushroomobserver.org/collection_numbers/9528
https://mushroomobserver.org/collection_numbers/9679
https://mushroomobserver.org/collection_numbers/9844
https://mushroomobserver.org/collection_numbers/9845
https://mushroomobserver.org/collection_numbers/9861
https://mushroomobserver.org/collection_numbers/9883
https://mushroomobserver.org/collection_numbers/9888
https://mushroomobserver.org/collection_numbers/9920
https://mushroomobserver.org/collection_numbers/9923
https://mushroomobserver.org/collection_numbers/9924
https://mushroomobserver.org/collection_numbers/9925
https://mushroomobserver.org/collection_numbers/9930
https://mushroomobserver.org/collection_numbers/9943
https://mushroomobserver.org/collection_numbers/9949
https://mushroomobserver.org/collection_numbers/9953
https://mushroomobserver.org/collection_numbers/9954
https://mushroomobserver.org/collection_numbers/9964
https://mushroomobserver.org/collection_numbers/9998
https://mushroomobserver.org/collection_numbers/9999
https://mushroomobserver.org/collection_numbers/10020
https://mushroomobserver.org/collection_numbers/10039
https://mushroomobserver.org/collection_numbers/10082
https://mushroomobserver.org/collection_numbers/10181
https://mushroomobserver.org/collection_numbers/10225
https://mushroomobserver.org/collection_numbers/10267
https://mushroomobserver.org/collection_numbers/10317
https://mushroomobserver.org/collection_numbers/10399
https://mushroomobserver.org/collection_numbers/10416
https://mushroomobserver.org/collection_numbers/10556
https://mushroomobserver.org/collection_numbers/10719
https://mushroomobserver.org/collection_numbers/10728
https://mushroomobserver.org/collection_numbers/10747
https://mushroomobserver.org/collection_numbers/10850
https://mushroomobserver.org/collection_numbers/11049
https://mushroomobserver.org/collection_numbers/11067
https://mushroomobserver.org/collection_numbers/11182
https://mushroomobserver.org/collection_numbers/11245
https://mushroomobserver.org/collection_numbers/11288
https://mushroomobserver.org/collection_numbers/11534
https://mushroomobserver.org/collection_numbers/11671
https://mushroomobserver.org/collection_numbers/11756
https://mushroomobserver.org/collection_numbers/11931
https://mushroomobserver.org/collection_numbers/11957
https://mushroomobserver.org/collection_numbers/13062
https://mushroomobserver.org/collection_numbers/8
https://mushroomobserver.org/collection_numbers/2282
https://mushroomobserver.org/collection_numbers/3123
https://mushroomobserver.org/collection_numbers/3124
https://mushroomobserver.org/collection_numbers/3133
https://mushroomobserver.org/collection_numbers/6354
https://mushroomobserver.org/collection_numbers/7601
https://mushroomobserver.org/collection_numbers/9859
https://mushroomobserver.org/collection_numbers/9863
https://mushroomobserver.org/collection_numbers/9967
https://mushroomobserver.org/collection_numbers/10252
https://mushroomobserver.org/collection_numbers/10269
https://mushroomobserver.org/collection_numbers/10273
https://mushroomobserver.org/collection_numbers/10341
https://mushroomobserver.org/collection_numbers/10555
https://mushroomobserver.org/collection_numbers/12851

**Comment by mo-nathan** [comment]
Looking through the CollectionNumbers with 3 observations, I found that a lot of them are different observations of the same taxon.  Often from the same day.  E.g., https://mushroomobserver.org/collection_numbers/10273.  Should these be different observations if they are going to share the same collection number?

**Comment by mo-nathan** [comment]
Now that I've dug deep into this, here's a proposed plan:

1. Fix the existing bug in CollectionNumbers where links on the Observation pages don't work.
2. Rename FieldSlips to CollectionCodes.
3. Add a message to the CollectionNumbers pages indicating that they are going to get deprecated in favor of CollectionCodes and Projects and pointing to an Article that describes this plan and asks folks to email webmaster@mushroomobserver.org if they have any questions and concerns.
4. Move forward with the CollectionCode concept using QR codes.
5. Create a UI on the Observation page that makes it at least as easy to create appropriate CollectionCodes as it currently is to create a CollectionNumber.  I think it would make sense to automatically select any project they are part of whose constraints are met by the observation.  That might get a little annoying for someone who is a member of a bunch of projects that are in some way mutually exclusive, but I think we cross that bridge after we get the select all working and when someone complains about it.
6. Create a way to create a Project based on that specific CollectionNumber or the CollectionNumber.name.  For cases where there is only one observation for a CollectionNumber it gets the code "<name> <number>".  For cases where there is more than one observation for a CollectionNumber it gets the code "<name> <number> <#>" where the numbers start at 1 and increment.
7. Update the deprecation warnings to point folks to that tool.  This might need to be a background job (the largest case would be a project with 1205 observations for "Terri Clements/Donna Fulton").
8. After some time, disable the CollectionNumber UIs and convert any remaining CollectionNumbers to CollectionCodes following the above algorithm.
9. Remove the CollectionNumber code and table.

**Comment by mo-nathan** [comment]
After all the above analysis and yet more thinking, I've concluded that Collection Numbers and Fields Slips actually serve different purposes and should be kept separate.  The key differentiators are:
1. Collection Numbers are manually created and managed and give their creators the ability to change them and use them as they see fit.
2. Field Slips are inherently tied to a pre-printed piece of paper that is not intended to change (but could be re-printed if it got destroyed).  This means users should *not* be able to create them in any way other than reading a QR code.
3. Field Slips must be more easily findable than Collection Numbers.  Specifically the full code of a Field Slip must be unique.
4. Collection Numbers are not as easily findable.  There is no simple way to search for them since they are potentially ambiguous.  Neither the "name" or "number" of a Collection Number are unique and their concatenation is not guaranteed to be unique.  For example, you can create a collection number with a name of "Nathan Wilson" and a number of "1", and you could create a collection number with a name of "Nathan" and a number of "Wilson 1".

**Comment by JoeCohen** [comment]
(Adding this here so it doesn't get lost in email.)
Joseph Cohen <jdcohenesq@gmail.com>
Field Slip Form suggestions
Joseph Cohen <jdcohenesq@gmail.com>
To: Nathan Wilson <nathan@collectivesource.com>
Cc: Deana Tempest <deanatempest@gmail.com>
Nathan:
Thanks for the response.
Interesting point about Date. I agree it's relatively unimportant. But it could matter when someone is doing a search with a Date parameter. (And it will matter to OCD users.)
FYI. It takes me 3-6 minutes per slip to digitize the Field Slips which Deana shared with me.
She shared 191 slips.
I've done 30 so far.
So about 8-16 hours of work remaining. I anticipate spreading that out over many days.
-Joe
>
> On Sat, Jan 11, 2025 at 7:50AM Nathan Wilson <nathan@collectivesource.com> wrote:
Yeah, I think some of that stuff is supposed to already be sticky, but I've seen it behave oddly. Haven't dug into it, but I agree with the direction. Deriving the date from the location is tricky a
locations before or after the official walk and after the fact I don't think anyone really cares about the date beyond when the entire event happens. I also think having the location be meanin
the Field Slip form and have it look up the associated Location. I suppose the Project Alias could optionally include a date in the case of Location Aliases. The tricky part then becomes dec
date also potentially interacts with any implied date for the location. I wonder if the date should by default fit into any project constraint. Anyway, there's some trickiness here.

>> On Fri, Jan 10, 2025 at 6:30PM Joseph Cohen <jdcohenesq@gmail.com> wrote:
Nathan:
TLDR:
>> 1. Make the Location sticky ( (I.e., the last Field Slip Location entered by the user, or at least the last Field Slip Location entered by the user in the last hour).
>> 2. Make the Date sticky. Also see 3.
>> 3. Move Location before Date, and derive Date from the Location
Explanation.
I'm beginning to create Field Slips from the Photos Album which Deanna shared with me last night. I'm using the workflow detailed below. There are a few steps that could be saved by changing the Field Slip Code.
>>> 1. The Location always starts as Walk 7. Deana's photos are often grouped by walk. If the Location were sticky, I could avoid some re-typing.
>>> 2. The Date always starts as today's date. The photos are grouped by date. So a sticky date would always save me the trouble of changing the year, and most of the time would save
>>> 3. The date can be derived from the Location (for the real walks/forays). So I suggest putting the Location first, then deriving the Date if possible. This might imply a change in the printed Field Slips.

>> Joe’s workow for adding Field Slips based on photos in Deana’s Shared Album, NEMF 2024:
Connect a QR Code scanner to computer (optional)
>> - Open NEMF Collector Cleanup in a window on left side s of screen
>> - Open Photos, Shared Album in a window on left side of screen, partly covering Collector Cleanup
>> - Open NEMF Identifier Cleanup on right side of screen.
>> - Open MO in a window on right side of screen, partly covering Identifier Cleanup
>> 
>> Resulting Windows:
![image](https://github.com/user-attachments/assets/8df02374-e253-4aec-9966-446f8abee085)
>> - In the Photos window, double-click a Field Slip
>> - In the MO window click the QR Code icon
>> - Scan the Field Slip QR code
>> - When the url appears, Enter
Result: Record Field Slip Form
>> - Enter the Date
>> - Enter Collector (referring to Collector Cleanup spreadsheet)
>> - Enter the Walk
>> - Enter ID
>> - Enter ID By (referring to Identifier Cleanup spreadsheet)
>> - Enter Notes
>> - Quick Create
>> - Result:
Observation without images
>> - Edit the Observation
>> - Drag the Photo onto the Edit Observation tab
>> - Edit Copyright Holder
>> - Save Edits.

**Comment by JoeCohen** [comment]
The order of items in the form page should be the same as their order on the paper form.
This would decrease visual/mental  discordance, make for a smoother workflow, make it harder to forget items.

Ciurrent Order:
| paper form | form page |
| ---------- | --------- |
| Collector | Date |
| Date | Collector |
| Location | Location |
| Notes | ID |
| ID | ID By |
| n.a. | Quick Create |
| ID By |  Notes | 
| n.a. | suggest another Create button |

**Comment by mo-nathan** [comment]
Implemented.
